### PR TITLE
Fixed conflicts with PHP 8.2, added type hinting for properties and…

### LINKED ISSUE #52: Fix PHP 8.2 deprecations
[This customer](https://wordpress.org/support/topic/deprecated-creation-of-dynamic-property-16/) reported some PHP 8.2 deprecations in Menu Cart.

### PR DESCRIPTION
**Pull request by brualio**
…refactoring.

Closes #52

### CODE REVIEW

File: includes/wpmenucart-edd.php [hunk 1]
```diff
@@ -2,20 +2,20 @@
...
-	    /**
-	     * Construct.
-	     */
-	    public function __construct() {
-	    }
+		/**
+		 * Construct.
+		 */
+		public function __construct() {
+		}
```
Comment (MohamadNateqi) [inline includes/wpmenucart-edd.php:9]: This can be removed.

File: includes/wpmenucart-edd.php [hunk 2]
```diff
@@ -2,20 +2,20 @@
...
 	
 		public function menu_item() {
 			global $post;
```
Comment (MohamadNateqi) [inline includes/wpmenucart-edd.php:12]: This seems unused, and it would be good to remove it.

File: includes/wpmenucart-settings.php [hunk 3]
```diff
@@ -402,21 +397,24 @@ public function init_settings() {
...
+		}
 
 		// Convert old wpmenucart menu settings to array
-		if ( isset($option_values['menu_name_1']) ) {
-			$option_values['menu_slugs'] = array( '1' =>  $option_values['menu_name_1'] );
-			update_option( 'wpmenucart', $option_values );
+		if ( isset( $option_values['menu_name_1'] ) ) {
+			$option_values['menu_slugs'] = array(
+				'1' => $option_values['menu_name_1']
+			);
```
Comment (MohamadNateqi) [inline includes/wpmenucart-settings.php:408]: Why multiple lines?

File: includes/wpmenucart-settings.php [hunk 4]
```diff
@@ -454,33 +452,36 @@ public function wpmenucart_admin_styles() {
 	 * Default settings.
 	 */
 	public function default_settings() {
-		$wcmenucart_options = get_option('wcmenucart');
-		$menu_slugs = array( '1' =>  isset($wcmenucart_options['menu_name_1']) ? $wcmenucart_options['menu_name_1']:'0' );
+		$option              = 'wcmenucart';
+		$wcmenucart_options  = get_option( $option, array() );
+		$menu_slugs          = array(
+			'1' => isset( $wcmenucart_options['menu_name_1'] ) ? $wcmenucart_options['menu_name_1'] : '0'
```
Comment (MohamadNateqi) [inline includes/wpmenucart-settings.php:458]: WPCS - [For an array with elements in multiple lines, adding a comma at the end is recommended.](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#:~:text=Note%20the%20comma%20after%20the%20last%20array%20item%3A%20this%20is%20recommended%20because%20it%20makes%20it%20easier%20to%20change%20the%20order%20of%20the%20array%2C%20and%20makes%20for%20cleaner%20diffs%20when%20new%20items%20are%20added.)

File: wp-menu-cart.php [hunk 5]
```diff
@@ -135,10 +172,10 @@ public function load_classes() {
 	 */
 	public function good_to_go() {
 		$wpmenucart_shop_check = get_option( 'wpmenucart_shop_check' );
-		$active_plugins = $this->get_active_plugins();
+		$active_plugins        = $this->get_active_plugins();
```
Comment (MohamadNateqi) [inline wp-menu-cart.php:175]: This is an unused variable, and it would be good to remove it.

**Review by MohamadNateqi — CHANGES_REQUESTED** [review]

**Review by BrunoPavlinic98 — APPROVED** [review]
LGTM!

**Pull request merged**